test/plugins/Library/io/serial/YamlMapTest.java

```diff
@@ -29,38 +24,34 @@
 public class YamlMapTest extends TestCase {
 
 	public void testYamlMap() throws IOException {
-		Map<String, Object> data = new TreeMap<String, Object>();
+		Map<String, Object> data = new TreeMap<>();
 		data.put("key1", new Bean());
 		data.put("key2", new Bean());
 		data.put("key3", new Custom("test"));
 		data.put("key4", new Wrapper("test", new Custom("test")));
 
-		Yaml yaml = new Yaml(new Loader(new ExtendedConstructor()),
-			                new Dumper(new ExtendedRepresenter(), new DumperOptions()));
+		Yaml yaml = new Yaml(new ExtendedConstructor(), new ExtendedRepresenter(), new DumperOptions());
 		File file = new File("beantest.yml");
 
 		FileOutputStream os = new FileOutputStream(file);
 		yaml.dump(data, new OutputStreamWriter(os));
 		os.close();
 
 		FileInputStream is = new FileInputStream(file);
-		Object o = yaml.load(new InputStreamReader(is));
+		Map<String, Object> map = yaml.load(new InputStreamReader(is));
 		is.close();
 
-		assertTrue(o instanceof Map);
-		Map m = (Map)o;
-		assertTrue(m.get("key1") instanceof Bean);
-		assertTrue(m.get("key2") instanceof Bean); // NOTE these tests fail in snakeYAML 1.2 and below, fixed in 1.3
-		assertTrue(m.get("key3") instanceof Custom);
-		assertTrue(m.get("key4") instanceof Wrapper);
+		assertTrue(map.get("key1") instanceof Bean);
```

Comment:
This only tests for the general existence in the map but does not validate the values… at least for `key4` I would expect a bit more here. 😃